### PR TITLE
Check for symlinks prior to counting file as a hardlink

### PR DESCRIPTION
**Pull request by glicholas**
## Description

When counting existing hardlinks of a file to determine if all contributing hardlinks are within the root_dir (which won't count), get_inode_count was also counting symlinks towards the total count.  This now skips symlinks so they do not count.

Fixes # (608)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated the docstring for new or existing methods
- [x] I have modified this PR to merge to the develop branch

### CODE REVIEW

File: modules/util.py [hunk 1]
```diff
@@ -536,22 +536,28 @@ def __init__(self, root_dir, remote_dir):
...
-                continue
-            except Exception as ex:
-                logger.stacktrace()
-                logger.error(ex)
-                continue
-            if inode_no in self.inode_count:
-                self.inode_count[inode_no] += 1
-            else:
-                self.inode_count[inode_no] = 1
+            # Only check hardlinks for files
```
Comment (bakerboy448) [inline modules/util.py:539]: Can we shorten this to a single if?

if is file AND is not link then continue?

Comment (glicholas) [reply modules/util.py:539]: Yeah, that works too.  Originally I thought islink() would throw an exception for directories, but that's not true.

**Review by bobokun — APPROVED** [review]
